internal/datastore/postgres/postgres.go

```diff
@@ -146,6 +146,19 @@ func newPostgresDatastore(
 		return nil, common.RedactAndLogSensitiveConnString(ctx, errUnableToInstantiate, err, pgURL)
 	}
 
+	// Setup the credential provider
+	credentialProvider, err := datastore.NewCredentialProvider(ctx, config.credentialProviderName)
+	if err != nil {
+		return nil, err
+	}
+	if !credentialProvider.IsDynamic() {
```

Comment:
Alternatively we could make `NewCredentialProvider` return `nil`, or just not call it if `credentialProviderName` is empty.